phoenix-core/src/main/java/org/apache/phoenix/iterate/ClientHashSizeException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.iterate;
+
+/**
+ * Thrown by {@link org.apache.phoenix.iterate.ClientHashAggregatingResultIterator } when
+ * hash size exceeds memory threshold.
+ * 
+ */
+public class ClientHashSizeException extends RuntimeException {
```

Comment:
You won't need this as an InsufficientMemoryException will be thrown if you go above the specified memory limit (based on existing Phoenix config properties) and this will be unwound to become a SQLException with the code SQLExceptionCode.INSUFFICIENT_MEMORY.

Reply:
I left this in as an additional check that uses SPOOL_THRESHOLD_BYTES. So both the MemoryManager and SPOOL_THRESHOLD_BYTES are enforced, in that order. I hope this is ok.